Applications/Spire/Source/Ui/Checkbox.cpp

```diff
@@ -0,0 +1,129 @@
+#include "Spire/Ui/Checkbox.hpp"
+#include <QEvent>
+#include <QFontMetrics>
+#include <QMouseEvent>
+#include <QPainter>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+using namespace Spire;
+
+namespace {
+  auto BOX_SIZE() {
+    static auto size = scale(16, 16);
+    return size;
+  }
+
+  auto PADDING() {
+    static auto padding = scale_width(8);
+    return padding;
+  }
+}
+
+Checkbox::Checkbox(QWidget* parent)
+  : QCheckBox("", parent) {}
+
+Checkbox::Checkbox(const QString& label, QWidget* parent)
+    : QCheckBox(label, parent),
+      m_is_read_only(false),
+      m_check_icon(imageFromSvg(":/Icons/check.svg", scale(10, 8))) {
+  setAttribute(Qt::WA_Hover);
+  auto font = QFont("Roboto");
+  font.setPixelSize(scale_height(12));
+  setFont(font);
+}
+
+void Checkbox::keyPressEvent(QKeyEvent* event) {
+  event->accept();
+  if(m_is_read_only || event->isAutoRepeat()) {
+    return;
+  }
+  if(event->key() == Qt::Key_Space) {
+    setChecked(!isChecked());
```

Comment:
QCheckBox doesn't ```update()```?

Also is this duplicating work done by the ```QCheckBox```? Can you delegate to ```QCheckBox:::keyPressEvent```?